dotnet/Vaas/src/Vaas/Vaas.cs

```diff
@@ -183,6 +183,68 @@ public async Task<VaasVerdict> ForUrlAsync(Uri uri, CancellationToken cancellati
         return new VaasVerdict(verdictResponse);
     }
 
+    public async Task<VaasVerdict> ForStreamAsync(
+        Stream stream,
+        CancellationToken cancellationToken,
+        Dictionary<string, string>? verdictRequestAttributes = null
+    )
+    {
+        if (stream == null)
+            throw new VaasClientException("Stream was null.");
+
+        var verdictResponse = await ForStreamRequestAsync(
+            new VerdictRequestForStream(SessionId ?? throw new InvalidOperationException())
+            {
+                UseCache = _options.UseCache,
+                UseHashLookup = _options.UseHashLookup,
+                VerdictRequestAttributes = verdictRequestAttributes
+            });
+        if (!verdictResponse.IsValid)
+            throw new JsonException("VerdictResponse is not valid");
+        if (verdictResponse.Verdict != Verdict.Unknown)
+            throw new VaasServerException("Server returned verdict without receiving content.");
+
+        if (
+            string.IsNullOrWhiteSpace(verdictResponse.Url)
+            || string.IsNullOrWhiteSpace(verdictResponse.UploadToken)
+        )
+        {
+            throw new JsonException(
+                "VerdictResponse missing URL or UploadToken for stream upload."
+            );
+        }
+
+        var response = WaitForResponseAsync(verdictResponse.Guid);
+        await UploadStream(stream, verdictResponse.Url, verdictResponse.UploadToken, cancellationToken);
+        
+        return new VaasVerdict(await response);
+    }
+    
+    private async Task UploadStream(Stream stream, string url, string token, CancellationToken cancellationToken)

```

Comment:
I noticed that this function is very similar with this function here https://github.com/GDATASoftwareAG/vaas/blob/d975d583ee41bfa549c8591ab3a47741e21899a6/dotnet/Vaas/src/Vaas/Vaas.cs#L335
Any way to combinate these two since both are working with Streams?